### ISSUE TITLE
Timeout for algorithmicx+cleveref

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#1157]
Encountered when diagnosing arXiv, here is a minimally reduced example for this infinite loop:
```tex
\documentclass{article}

\usepackage{algorithm}
\usepackage{algpseudocode}
\usepackage{hyperref}
\usepackage{cleveref}

\begin{document}

\begin{algorithm}
	\begin{algorithmic}
		\Function{RandomBFS}{$G, v, q${}}
		\EndFunction
	\end{algorithmic}
\end{algorithm}

\end{document}
```

Don't have time to fish for a patch immediately, so bookkeeping here first.

**Comment by dginev** [comment]
FYI I stumbled on two separate examples of this timeout in the arxiv_1905 monthly test corpus. Anecdotal, but already indicating there is _some_ number of timeouts in converting arxiv associated with this pattern.

**Comment by brucemiller** [comment]
So the fatal should be fixed with 8bdff7f3ef7a72ad9879c204305b8ac0cc148b94, hopefully; of course it turns into an error.

 The underlying problem seems to be an interaction between algorithm, hyperref and cleveref, with cleveref changing its behaviour based on the other 2 being loaded.  The tricky bit seems to be that it's deep into hyperref's guts and that isn't being reflected in latexml.  Not sure that this is easily fixable in the short term. Could close or save for another day.

**Comment by dginev** [comment]
If the timeout is fixed, we can close and re-file when the error becomes "important enough", since we're trying to get the biggest impact items first. I mostly cared about removing some timeouts, to speedup the corpus conversion and get more HTML documents out. Thanks for the fix!
